portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -51,13 +54,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    // Get input from the form.
+    // Set commenter email to email from UserService if they opted to show their email and Unknown
+    // otherwise.
+    UserInfo userInfo = UserInfoUtils.getUserInfo(userService.getCurrentUser().getUserId());
+    String commenterEmail;
+    if (userInfo == null) {
+      commenterEmail = "Unknown";
+    } else {
+      String userEmail = userInfo.getEmail();
+      commenterEmail = userInfo.getShowEmail() ? userEmail : "Unknown";
+    }
```

Comment:
can be a separate method, here can just be commenterEmail = getCommenterEmail

nit: the logic can be simpler:

if (userInfo != null && userInfo.GetShowEmail()) {
return useremail.
}
return "Unknown"

Reply:
Done - fixed logic.